src/client/components/MapHeader.tsx

```diff
@@ -160,7 +163,14 @@ const MapHeader = ({
   readonly selectedGeounits: GeoUnits;
   readonly advancedEditingEnabled?: boolean;
   readonly isReadOnly: boolean;
+  readonly limitSelectionToCounty: boolean;
 }) => {
+  const [topGeoLevelName, setTopGeoLevelName] = useState<string | undefined>(undefined);
+  useEffect(() => {
+    if (metadata) {
+      setTopGeoLevelName(metadata.geoLevelHierarchy[metadata.geoLevelHierarchy.length - 1].id);
+    }
+  }, [metadata]);
```

Comment:
So we don't actually need to introduce more component-level state or effects here. We really just want a value that's derived from `metadata` so we can just assign a variable:
```suggestion
  const topGeoLevelName = metadata
    ? metadata.geoLevelHierarchy[metadata.geoLevelHierarchy.length - 1].id
    : undefined;
```
`topGeoLevelName` will be updated whenever `metadata` is since the component will re-render whenever `metadata` changes.

In general we should try to avoid introducing state/effects whenever possible (our ideal components are just stateless render functions that take in props as input and return JSX since that keeps them as easy to reason about as possible).